### ISSUE TITLE
19.03 - Containers DOWN

### ISSUE THREAD

**Issue by alphaDev23** [dtag-dev-sec/tpotce#658]
See below for basic info.  ALL containers DOWN.

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
[tsec@tpot-2:~]$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux bullseye/sid
Release:	unstable
Codename:	sid

 uname -a
Linux tpot-2 5.6.0-2-amd64 #1 SMP Debian 5.6.14-2 (2020-06-09) x86_64 GNU/Linux

- What T-Pot version are you currently using? 19.03
- What edition (Standard, Nextgen, etc.) of T-Pot are you running? Standard
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?  VM
- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.  Nothing obvious in install.log.  install.err:
[tsec@tpot-2:~]$ cat /install.err 
info: Trying to set 'docker.io/restart' [boolean] to 'true'
info: Loading answer for 'docker.io/restart'
info: Trying to set 'debconf/frontend' [select] to 'noninteractive'
info: Loading answer for 'debconf/frontend'
W: --force-yes is deprecated, use one of the options starting with --allow instead.
Extracting templates from packages: 100%
W: --force-yes is deprecated, use one of the options starting with --allow instead.
E: Sub-process /usr/bin/dpkg returned an error code (1)
Adding password for user litmusblue
Generating a RSA private key
....+++
......................................+++
writing new private key to '/data/nginx/cert/nginx.key'
-----
npm WARN npm npm does not support Node.js v10.15.2
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN tar write after end
npm WARN tar write after end
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-08-01T23_24_42_792Z-debug.log
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
fatal: destination path '/opt/tpot' already exists and is not an empty directory.
Created symlink /etc/systemd/system/multi-user.target.wants/tpot.service → /etc/systemd/system/tpot.service.
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-4.19.0-5-amd64
Found initrd image: /boot/initrd.img-4.19.0-5-amd64
Found linux image: /boot/vmlinuz-4.9.0-9-amd64
Found initrd image: /boot/initrd.img-4.9.0-9-amd64
done
Trying: dig +short myip.opendns.com @resolver4.opendns.com
Trying: dig +short -4 -t a whoami.akamai.net @ns1-1.akamaitech.net
Trying: dig +short myip.opendns.com @resolver2.opendns.com
Trying: dig +short whoami.akamai.net @ns1-1.akamaitech.net
Trying: dig +short myip.opendns.com @resolver1.opendns.com
Trying: dig +short myip.opendns.com @resolver3.opendns.com
Trying: curl -s ipecho.net/plain

- How long has your installation been running?  1 day
- Did you install upgrades, packages or use the update script?  update script
- Did you modify any scripts or configs? If yes, please attach the changes. No
- Please provide a screenshot of `glances` and `htop`.
- How much free disk space is available (`df -h`)?
[tsec@tpot-2:~]$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            7.9G     0  7.9G   0% /dev
tmpfs           1.6G  1.1M  1.6G   1% /run
/dev/vda1        16G  7.9G  7.1G  53% /
tmpfs           7.9G     0  7.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           7.9G     0  7.9G   0% /sys/fs/cgroup
tmpfs           1.6G     0  1.6G   0% /run/user/1000

- What is the current container status (`dps.sh`)?
$ sudo /opt/tpot/bin/dps.sh
========| System |========
    Date:  Tue 23 Jun 2020 02:14:27 PM UTC

  Uptime:  14:14:34 up 10:38,  2 users,  load average: 10.06, 7.57, 7.42

NAME                  STATUS                       PORTS
adbhoney              DOWN                  
ciscoasa              DOWN                  
conpot_guardian_ast   DOWN                  
conpot_iec104         DOWN                  
conpot_ipmi           DOWN                  
conpot_kamstrup_382   DOWN                  
cowrie                DOWN                  
cyberchef             DOWN                  
dionaea               DOWN                  
elasticpot            DOWN                  
elasticsearch         DOWN                  
ewsposter             DOWN                  
head                  DOWN                  
heralding             DOWN                  
honeytrap             DOWN                  
kibana                DOWN                  
logstash              DOWN                  
mailoney              DOWN                  
medpot                DOWN                  
nginx                 DOWN                  
p0f                   DOWN                  
rdpy                  DOWN                  
snare                 DOWN                  
spiderfoot            DOWN                  
suricata              DOWN                  
tanner                DOWN                  
tanner_api            DOWN                  
tanner_phpox          DOWN                  
tanner_redis          DOWN                  
tanner_web            DOWN      
- What is the status of the T-Pot service (`systemctl status tpot`)?
tsec@tpot-2:~]$ sudo systemctl status tpot
● tpot.service - tpot
     Loaded: loaded (/etc/systemd/system/tpot.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2020-06-23 14:14:00 UTC; 55s ago
    Process: 552233 ExecStartPre=/opt/tpot/bin/updateip.sh (code=exited, status=0/SUCCESS)
    Process: 552590 ExecStartPre=/bin/bash -c /opt/tpot/bin/clean.sh on (code=exited, status=0/SUCCESS)
    Process: 552796 ExecStartPre=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml down -v (code=exited, status=1/FAILURE)
    Process: 553439 ExecStartPre=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml rm -v (code=exited, status=0/SUCCESS)
    Process: 553441 ExecStartPre=/bin/bash -c docker network rm $(docker network ls -q) (code=exited, status=1/FAILURE)
    Process: 553764 ExecStartPre=/bin/bash -c docker volume rm $(docker volume ls -q) (code=exited, status=1/FAILURE)
    Process: 553776 ExecStartPre=/bin/bash -c docker rm -v $(docker ps -aq) (code=exited, status=1/FAILURE)
    Process: 553789 ExecStartPre=/bin/bash -c docker rmi $(docker images | grep "<none>" | awk '{print $3}') (code=exited, status=1/FAILURE)
    Process: 553804 ExecStartPre=/bin/bash -c /sbin/ethtool --offload $(/sbin/ip address | grep "^2: " | awk '{ print $2 }' | tr -d [:punct:]) rx off tx off (code=exited, status=1/FAILURE)
    Process: 553810 ExecStartPre=/bin/bash -c /sbin/ethtool -K $(/sbin/ip address | grep "^2: " | awk '{ print $2 }' | tr -d [:punct:]) gso off gro off (code=exited, status=0/SUCCESS)
    Process: 553816 ExecStartPre=/bin/bash -c /sbin/ip link set $(/sbin/ip address | grep "^2: " | awk '{ print $2 }' | tr -d [:punct:]) promisc on (code=exited, status=0/SUCCESS)
    Process: 553822 ExecStartPre=/opt/tpot/bin/rules.sh /opt/tpot/etc/tpot.yml set (code=exited, status=0/SUCCESS)
   Main PID: 553890 (docker-compose)
      Tasks: 47 (limit: 19184)
     Memory: 43.2M
     CGroup: /system.slice/tpot.service
             └─553890 /usr/bin/python3 /usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml up --no-color

Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating tanner_redis        ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating tanner_phpox        ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating cyberchef           ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating mailoney            ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating conpot_ipmi         ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating adbhoney            ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating suricata            ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating p0f                 ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating conpot_iec104       ...
Jun 23 14:14:15 tpot-2 docker-compose[553890]: Creating medpot              ...

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
[tsec@tpot-2:~]$ sudo systemctl stop tpot
[tsec@tpot-2:~]$ netstat -tulpen
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode      PID/Program name    
tcp        0      0 127.0.0.1:61209         0.0.0.0:*               LISTEN      0          16335      -                   
tcp        0      0 0.0.0.0:64295           0.0.0.0:*               LISTEN      0          16604      -                   
tcp6       0      0 :::64294                :::*                    LISTEN      0          14460      -                   
tcp6       0      0 :::64295                :::*                    LISTEN      0          16615      -                   
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          14286      -                   
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          14099      -                   
udp        0      0 192.168.1.230:123       0.0.0.0:*                           0          16073      -                   
udp        0      0 127.0.0.1:123           0.0.0.0:*                           0          16071      -                   
udp        0      0 0.0.0.0:123             0.0.0.0:*                           0          16067      -                   
udp6       0      0 :::123                  :::*                                0          16064      -     
- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries
ALL the containers are DOWN

**Comment by t3chn0m4g3** [comment]
> fatal: destination path '/opt/tpot' already exists and is not an empty directory.

From what I can see either the installer was run more than once or the `/opt/tpot` folder already existed. This needs a complete fresh install.

**Comment by alphaDev23** [comment]
Is there a way to do a fresh install with the current instance?  I had installed tpot, then uploaded the image to Openstack, and then launched the instance from Horizon. I was expecting the instance to come up without issue such that a new instance would not require a re-install.  

Also, what is the reason that the above method is not working?

As a note, I do see an Ansible Openstack implementation but creating one image with tpot installed and then launching may be more efficient.

**Comment by t3chn0m4g3** [comment]
Only a fresh install can solve it.

**Comment by alphaDev23** [comment]
Why are the containers being recreated/reinitialized?  I just logged into Kibana, everything looked fine.  Then I logged into the console under the tsec user to check the containers.  Containers were up for 9 hours which would mean that the containers were running as expected for that time.  30 minutes later, the containers re-created themselves, exited, etc.

Is there a script that is being run when a user logs in under tsec?
